### PR TITLE
bpo-31341: cleanup IRIX-specific macros sgi and __sgi

### PR DESCRIPTION
**Pull request by yan12125**
To make work on curses-related modules simpler https://bugs.python.org/issue28190

https://bugs.python.org/issue31341

### CODE REVIEW

File: Modules/_cursesmodule.c [hunk 1]
```diff
@@ -112,7 +112,7 @@ static const char PyCursesVersion[] = "2.2";
 #define CURSES_MODULE
 #include "py_curses.h"
 
-#if defined(HAVE_TERM_H) || defined(__sgi)
+#if defined(HAVE_TERM_H)
```
Comment (benjaminp) [inline Modules/_cursesmodule.c]: Can you change this to `#ifdef`?

File: Modules/posixmodule.c [hunk 2]
```diff
@@ -6401,8 +6391,7 @@ os_sched_getaffinity_impl(PyObject *module, pid_t pid)
 
 
 /* AIX uses /dev/ptc but is otherwise the same as /dev/ptmx */
-/* IRIX has both /dev/ptc and /dev/ptmx, use ptmx */
-#if defined(HAVE_DEV_PTC) && !defined(HAVE_DEV_PTMX)
+#if defined(HAVE_DEV_PTC)
```
Comment (benjaminp) [inline Modules/posixmodule.c]: Change to `ifdef`.

**Review by benjaminp — CHANGES_REQUESTED** [review]
This looks mostly good but needs a rebase.

**Comment by yan12125** [comment]
I have made the requested changes; please review again.

As a side note, the merge conflict was due to additions for VxWorks (https://github.com/python/cpython/pull/12118/files#diff-a6f29e907cbb5fffd44d453bcd7b77d5R232).

**Comment by yan12125** [comment]
Most IRIX-specific codes are removed in https://github.com/python/cpython/pull/23023, while there are still references to IRIX-specific macros `sgi` and `__sgi`. I rebased this pull request to only remove remaining macros.

**Comment by yan12125** [comment]
Rebased for recent cleanups in Include/Python.h (https://bugs.python.org/issue45434)

**Comment by yan12125** [comment]
Seems no sign of further reviewing, closing.

**Pull request closed without merging**